catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/DmsCoordinateProcessor.java

```diff
@@ -0,0 +1,215 @@
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A regex-based processor for identifying potential DMS coordinate literals in a given input string
+ * despite deviations from the official spec. The general policy is to be as forgiving as possible
+ * without sacrificing determinism.
+ */
+public class DmsCoordinateProcessor {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DmsCoordinateProcessor.class);
+
+  private static final String DEGREES_LAT_REGEX_STRING = "90|[0-8]?[0-9]";
+  private static final String DEGREES_LON_REGEX_STRING = "180|1[0-7][0-9]|0?[0-9]?[0-9]";
+
+  private static final String MINUTES_REGEX_STRING = "[0-5]?[0-9]";
+  private static final String SECONDS_REGEX_STRING = "[0-5]?[0-9](?:\\.\\d*)?";
+
+  private static final String DMS_LAT_REGEX_STRING =
+      "(?:\\d+\\D|\\D)*?("
+          + DEGREES_LAT_REGEX_STRING
+          + ")\\D+("
+          + MINUTES_REGEX_STRING
+          + ")\\D+("
+          + SECONDS_REGEX_STRING
+          + ")\\D*([NnSs])\\D*";
+
+  private static final String DMS_LON_REGEX_STRING =
+      "(?:\\d+\\D|\\D)*?("
+          + DEGREES_LON_REGEX_STRING
+          + ")\\D+("
+          + MINUTES_REGEX_STRING
+          + ")\\D+("
+          + SECONDS_REGEX_STRING
+          + ")\\D*([EeWw])\\D*";
+
+  private static final int DEGREES_LAT_GROUP = 1;
+  private static final int MINUTES_LAT_GROUP = 2;
+  private static final int SECONDS_LAT_GROUP = 3;
+  private static final int DIRECTION_LAT_GROUP = 4;
+
+  private static final int DEGREES_LON_GROUP = 5;
+  private static final int MINUTES_LON_GROUP = 6;
+  private static final int SECONDS_LON_GROUP = 7;
+  private static final int DIRECTION_LON_GROUP = 8;
+
+  private static final Pattern PATTERN_DMS_COORDINATE =
+      Pattern.compile(DMS_LAT_REGEX_STRING + DMS_LON_REGEX_STRING);
+
+  // This key tells the UI that the geo is on the suggestion itself
+  private static final String LITERAL_SUGGESTION_ID = "LITERAL-DMS";
+
+  private static class CoordinateTranslator {
```

Comment:
✏️ I didn't even notice the support functions were in a nested class. We can get rid of this as it provides no added value. Even saying it "helps organize" things is not a compelling reason to do it. The two big reasons we use classes are for holding state or providing layers of indirection. This accomplishes neither. It doesn't even provide additional visibility protection to its members. 

In general, stuff like this is the beginning of "util" classes, which essentially means we use classes to "organize" functions without any data nearby to govern that organization. If our mission is to group functions with the data they operate on, "util" classes can certainly **become** an impediment to that goal. It's hard to reasonably constrain, validate, and enforce scope on the input to those functions because they eventually get called by everyone.

Reply:
✏️ I'd also say the fact this is named `CoordinateTranslator` is confusing because that is the name of a USNG interface. Another reason to remove this: we have less terminology to manage.